tla-io/src/main/scala/at/forsyte/apalache/io/json/TlaToJson.scala

```diff
@@ -188,13 +188,20 @@ class TlaToJson[T <: JsonRepresentation](
             "body" -> bodyJson,
         )
 
-      case TlaAssumeDecl(body) =>
+      case TlaAssumeDecl(definedName, body) =>
         val bodyJson = apply(body)
-        withLoc(
+        val fields = Array[(String, T)](
             typeFieldName -> typeTagPrinter(decl.typeTag),
             kindFieldName -> "TlaAssumeDecl",
             "body" -> bodyJson,
         )
+
+        val f = definedName match {
+          case None       => fields
+          case Some(name) => fields :+ ("name" -> name: (String, T))
```

Comment:
Is it critical that the name will appear after the body in the json?

Reply:
No!